x/emissions/migrations/v5/migrate_test.go

```diff
@@ -0,0 +1,319 @@
+	s.Require().Equal(paramsExpected.MaxTopForecastersToReward, params.MaxTopForecastersToReward)
+	s.Require().Equal(paramsExpected.MaxTopReputersToReward, params.MaxTopReputersToReward)
+	s.Require().True(paramsExpected.CreateTopicFee.Equal(params.CreateTopicFee), "%s!=%s", paramsExpected.CreateTopicFee, params.CreateTopicFee)
+	s.Require().Equal(paramsExpected.GradientDescentMaxIters, params.GradientDescentMaxIters)
+	s.Require().True(paramsExpected.RegistrationFee.Equal(params.RegistrationFee), "%s!=%s", paramsExpected.RegistrationFee, params.RegistrationFee)
+	s.Require().Equal(paramsExpected.DefaultPageLimit, params.DefaultPageLimit)
+	s.Require().Equal(paramsExpected.MaxPageLimit, params.MaxPageLimit)
+	s.Require().Equal(paramsExpected.MinEpochLengthRecordLimit, params.MinEpochLengthRecordLimit)
+	s.Require().Equal(paramsExpected.BlocksPerMonth, params.BlocksPerMonth)
+	s.Require().True(paramsExpected.PRewardInference.Equal(params.PRewardInference), "%s!=%s", paramsExpected.PRewardInference, params.PRewardInference)
+	s.Require().True(paramsExpected.PRewardForecast.Equal(params.PRewardForecast), "%s!=%s", paramsExpected.PRewardForecast, params.PRewardForecast)
+	s.Require().True(paramsExpected.PRewardReputer.Equal(params.PRewardReputer), "%s!=%s", paramsExpected.PRewardReputer, params.PRewardReputer)
+	s.Require().True(paramsExpected.CRewardInference.Equal(params.CRewardInference), "%s!=%s", paramsExpected.CRewardInference, params.CRewardInference)
+	s.Require().True(paramsExpected.CRewardForecast.Equal(params.CRewardForecast), "%s!=%s", paramsExpected.CRewardForecast, params.CRewardForecast)
+	s.Require().True(paramsExpected.CNorm.Equal(params.CNorm), "%s!=%s", paramsExpected.CNorm, params.CNorm)
+	s.Require().True(paramsExpected.EpsilonReputer.Equal(params.EpsilonReputer), "%s!=%s", paramsExpected.EpsilonReputer, params.EpsilonReputer)
+	s.Require().Equal(paramsExpected.HalfMaxProcessStakeRemovalsEndBlock, params.HalfMaxProcessStakeRemovalsEndBlock)
+	s.Require().True(paramsExpected.EpsilonSafeDiv.Equal(params.EpsilonSafeDiv), "%s!=%s", paramsExpected.EpsilonSafeDiv, params.EpsilonSafeDiv)
+	s.Require().True(paramsExpected.DataSendingFee.Equal(params.DataSendingFee), "%s!=%s", paramsExpected.DataSendingFee, params.DataSendingFee)
+	s.Require().Equal(paramsExpected.MaxElementsPerForecast, params.MaxElementsPerForecast)
+	s.Require().Equal(paramsExpected.MaxActiveTopicsPerBlock, params.MaxActiveTopicsPerBlock)
+	s.Require().Equal(paramsExpected.MaxStringLength, params.MaxStringLength)
+	s.Require().Equal(paramsExpected.InitialRegretQuantile, params.InitialRegretQuantile)
+	s.Require().Equal(paramsExpected.PNormSafeDiv, params.PNormSafeDiv)
+	s.Require().Equal(paramsExpected, params)
+}
+
+// in this test, we check that an already migrated topic, that only initialRegret
+// set to 0 but everything else will remain the same
+func (s *EmissionsV5MigrationTestSuite) TestMigratedTopicWithNaNInitialRegret() {
```

Comment:
I thought we had gotten rid of NaNs by now...?

Reply:
It's re-using the same functions as before because we still care to clear out fields. We're not clearing out NaNs necessarily but near-infinitely negative regrets that have developed, which shouldn't happen again once this PR is merged.